app/services/recipe_generator_service.rb

```diff
@@ -44,18 +44,28 @@ def prompt
     <<~CONTENT
       Write a recipe following these rules:
       1) The recipe MUST include only the ingredients provided.
-      2) Your response MUST be in JSON format, as this example:
+      2) Every preference MUST be incorporated into the recipe.
+      3) The recipe SHOULD exclude any ingredients listed as restrictions.
+      4) Your response MUST be in JSON format, as this example:
       { "name": "Dish Name",
         "content": "Recipe instructions" }
     CONTENT
   end
 
   def new_message
     [
-      { role: 'user', content: "Ingredients: #{message}" }
+      { role: 'user', content: "Ingredients: #{message}, Preferences: #{preferences}, Restrictions: #{restrictions}" }
     ]
   end
 
+  def restrictions
+    @user.preferences.where(restriction: true).map(&:description).join(', ') || ''
+  end
+
+  def preferences
+    @user.preferences.where(restriction: false).map(&:description).join(', ') || ''
+  end
```

Comment:
Also, both functions are the same except for restriction condition.

Maybe you can do something like this to avoid repetition:

```
def user_preferences(restriction)
  @user_preferences ||= @user.preferences.where(restriction: restriction).map(&:description).join(', ')
end

```
```suggestion
  def preferences
    user_preferences(false)
  end
  
  def restrictions
    user_preferences(true)
  end
```

Or maybe only one function but mapping like this: `"#{preference[:description]}, Restriction: #{preference[:restriction]}"`